### PR TITLE
CASMINST-5905

### PR DESCRIPTION
**Pull request by jpdavis-prof**
### Summary and Scope

Require one of **can-gateway** OR **chn-gateway** but do not allow both.
- Fixes: https://jira-pro.it.hpe.com:8443/browse/CASMINST-5905

#### Issue Type

- Bugfix Pull Request

### Prerequisites

- [x] I have included documentation in my PR (or it is not required)
- [x] I tested this on internal system (if yes, please include results or a description of the test)

### CODE REVIEW

File: cmd/init.go [hunk 1]
```diff
@@ -771,10 +779,26 @@ func validateFlags() []string {
 		}
 	}
 
+	userGateway := false
+	for i, gateway := range requiredUserGateway {
+		if v.IsSet(gateway) && v.GetString(gateway) != "" {
+			if userGateway {
+				errors = append(errors, fmt.Sprintf("provide only one of %v", requiredUserGateway))
```
Comment (rustydb) [inline cmd/init.go:786]: Nit: change error message to "Found value for both %v, expected only one or the other to be set."

**Review by rustydb — CHANGES_REQUESTED** [review]
Minor nit comment left.

Also, does this result in the CAN.conf or CHN.conf generation being omitted when the other is set? (e.g., when `--chn-gateway` is set, no `CAN.conf` is created)

File: cmd/init.go (hunk 1 above)
Comment (rustydb) [reply cmd/init.go:786]: The intent of my suggestion is to just make it more clear that CSI found definitions for both.
Like I said, it's a nit.

**Review by rustydb — APPROVED** [review]

**Comment by jpdavis-prof** [comment]
>Also, does this result in the CAN.conf or CHN.conf generation being omitted when the other is set? (e.g., when --chn-gateway is set, no CAN.conf is created)

It looks like when the CAN network is used, we create CAN.conf, but when the CHN network is used we do not create CHN.conf

**Comment by rustydb** [comment]
> > Also, does this result in the CAN.conf or CHN.conf generation being omitted when the other is set? (e.g., when --chn-gateway is set, no CAN.conf is created)
> 
> It looks like when the CAN network is used, we create CAN.conf, but when the CHN network is used we do not create CHN.conf

Right, looking more closely the CAN or CHN is chosen based on `--bican-user-network-name`.

**Pull request merged**